app/models/claim/matching_attribute_finder.rb

```diff
@@ -17,15 +15,26 @@ def initialize(source_claim, claims_to_compare = Claim.submitted)
     def matching_claims
       predicates = []
       values = []
-      ATTRIBUTES_TO_MATCH.each do |attribute|
-        value = @source_claim.read_attribute(attribute)
-        next if value.blank?
+      ATTRIBUTE_GROUPS_TO_MATCH.each do |attributes|
+        vals = values_for_attributes(attributes)
 
-        predicates << "LOWER(#{attribute}) = LOWER(?)"
-        values << value
+        next if vals.blank?
+
+        subject = "concat(#{attributes.join(",")})"
+
+        predicates << "LOWER(#{subject}) = LOWER(?)"
```

Comment:
We were wondering if these two variables could have more intention revealing names? It took a while for use to un-pick what was going on and we thought that might help? Something like: `columns_to_concat` and `comparison_statement`?